sql/src/test/java/io/crate/integrationtests/SQLTransportIntegrationTest.java

```diff
@@ -626,4 +647,34 @@ private boolean isSemiJoinsEnabled() {
             return false;
         }
     }
+
+    /**
+     * If the Test class or method contains a @UseRandomizedSession annotation then,
+     * based on the schema argument, a random (unquoted) schema name is returned. The schema name consists
+     * of a 1-20 character long ASCII string.
+     * For more details on the schema parameter see {@link io.crate.testing.UseRandomizedSession}
+     * <p>
+     * Method annotations have higher priority than class annotations.
+     */
+    private String RandomizedSchema() {
+        try {
+            Class<?> clazz = this.getClass();
+            Method method = clazz.getMethod(testName.getMethodName());
+            UseRandomizedSession annotation = method.getAnnotation(UseRandomizedSession.class);
+            if (annotation == null) {
+                annotation = clazz.getAnnotation(UseRandomizedSession.class);
+                if (annotation == null) {
+                    return Schemas.DOC_SCHEMA_NAME;
+                }
+            }
+            boolean randomize_schema = annotation.schema();
+            if (randomize_schema == false) {
+                return Schemas.DOC_SCHEMA_NAME;
+            }
+
+            return RandomStrings.randomAsciiOfLengthBetween(RandomizedContext.current().getRandom(), 1, 20).toLowerCase();
```

Comment:
@joemoe mentioned it could be valuable to generate a custom string that in addition to ASCII characters contains of numeric, `_`, characters etc. defined in [Naming Restrictions](https://crate.io/docs/crate/reference/en/latest/sql/ddl/basics.html#naming-restrictions) what do you say @chaudum @mfussenegger @mxm

Reply:
I'd keep that separated.

The main intention behind the schema randomization is to make sure we're always using the "defaultSchemaSession" as fallback, instead of implicitly using `doc`. 

The naming restrictions should be properly unit tested, but I don't think it's necessary to include it here. (Though it wouldn't hurt either, but I think the specific tests have a higher priority)

Reply:
I second @mfussenegger. I would only generate valid strings here and test invalid strings in a separate test.